app.rb

```diff
@@ -93,20 +97,101 @@
+      conformance_klass = FHIR::CapabilityStatement
+      supporting_resources = [
+        FHIR::AllergyIntolerance, FHIR::CarePlan, FHIR::CareTeam, FHIR::Condition, FHIR::Device,
+        FHIR::DiagnosticReport, FHIR::Goal, FHIR::Immunization, FHIR::MedicationRequest,
+        FHIR::MedicationStatement, FHIR::Observation, FHIR::Procedure, FHIR::RelatedPerson, FHIR::Specimen
+      ]
+      # Vital Signs includes these codes as defined in http://hl7.org/fhir/STU3/observation-vitalsigns.html
+      vital_signs = {
+        '85353-1' => 'Vital signs, weight, height, head circumference, oxygen saturation and BMI panel',
+        '9279-1' => 'Respiratory Rate',
+        '8867-4' => 'Heart rate',
+        '59408-5' => 'Oxygen saturation in Arterial blood by Pulse oximetry',
+        '8310-5' => 'Body temperature',
+        '8302-2' => 'Body height',
+        '8306-3' => 'Body height --lying',
+        '8287-5' => 'Head Occipital-frontal circumference by Tape measure',
+        '29463-7' => 'Body weight',
+        '39156-5' => 'Body mass index (BMI) [Ratio]',
+        '85354-9' => 'Blood pressure systolic and diastolic',
+        '8480-6' => 'Systolic blood pressure',
+        '8462-4' => 'Diastolic blood pressure'
+      }
+    end
+
+    # Parse accessible resources from scopes
+    accessible_resource_names = scopes.scan(/patient\/(.*?)\.read/)
+    accessible_resources = []
+    if accessible_resource_names.include?(["*"])
+      accessible_resources = supporting_resources.dup
+    else
+      accessible_resource_names.each do |w|
+        accessible_resources << Object.const_get("#{klass_header}#{w.first}")
+      end
+    end
 
-    patient_details = patient.massageHash(patient,true)
+    # Get the conformance statement
+    statement = client.conformance_statement
+    response.assert('Conformance Successfully Retrieved',statement.is_a?(conformance_klass),statement.fhirVersion)
+    statement_details = statement.to_hash
+
+    puts "FHIR Version: #{statement_details['fhirVersion']}"
+
+    # Get read capabilities
+    readable_resource_names = []
+    statement_details['rest'][0]['resource'].each do |r|
+      if r['interaction'].include?({"code"=>"read"})
+        readable_resource_names << "#{r['type']}"
+      end
+    end
+
+    # Get the patient demographics
+    patient = client.read(Object.const_get("#{klass_header}Patient"), patient_id).resource
+    response.assert('Patient Successfully Retrieved',patient.is_a?(Object.const_get("#{klass_header}Patient")),patient.id)
+    patient_details = patient.to_hash
     puts "Patient: #{patient_details['id']} #{patient_details['name']}"
 
     # DAF/US-Core CCDS
     response.assert('Patient Name',patient_details['name'],patient_details['name'])
-    response.assert('Patient Gender',FHIR::Patient::VALID_CODES[:gender].include?(patient_details['gender']),patient_details['gender'])
+    response.assert('Patient Gender',patient_details['gender'],patient_details['gender'])
```

Comment:
That's correct. When I attempted to check for a valid code this way using STU3, it raised an error. Since the rest of the patient extensions for DAF/US-Core are just based on whether it is null, I decided to switch it to test for this instead.